services/src/main/java/org/exoplatform/news/notification/provider/MailTemplateProvider.java

```diff
@@ -99,9 +106,61 @@ protected MessageInfo makeMessage(NotificationContext ctx) {
 
     @Override
     protected boolean makeDigest(NotificationContext ctx, Writer writer) {
-      return false;
+      EntityEncoder encoder = HTMLEntityEncoder.getInstance();
+      List<NotificationInfo> notifications = ctx.getNotificationInfos();
+      NotificationInfo notificationInfo = notifications.get(0);
+      try {
+        String spaceId = notificationInfo.getValueOwnerParameter(NotificationConstants.CONTENT_SPACE);
+        Space space = Utils.getSpaceService().getSpaceByDisplayName(spaceId);
+        if (!Utils.getSpaceService().isMember(space, notificationInfo.getTo())
+            || notificationInfo.getTo().equals(notificationInfo.getFrom())) {
+          return false;
+        }
+        String pluginId = notificationInfo.getKey().getId();
+        if (pluginId.equals(MentionInNewsNotificationPlugin.ID)) {
+          String mentionedIds = notificationInfo.getValueOwnerParameter(NotificationConstants.MENTIONED_IDS);
+          String ids = mentionedIds.substring(1, mentionedIds.length() - 1);
+          List<String> mentionedList = Stream.of(ids.split(",")).map(String::trim).collect(Collectors.toList());
+          if (!mentionedList.contains(notificationInfo.getTo())) {
+            return false;
+          }
+        }
+        String language = getLanguage(notificationInfo);
+        TemplateContext templateContext = new TemplateContext(pluginId, language);
+        //
+        Identity receiver = CommonsUtils.getService(IdentityManager.class).getOrCreateIdentity(OrganizationIdentityProvider.NAME, notificationInfo.getTo());
+        templateContext.put("FIRST_NAME", encoder.encode(receiver.getProfile().getProperty(Profile.FIRST_NAME).toString()));
+        templateContext.put("FOOTER_LINK", LinkProviderUtils.getRedirectUrl("notification_settings", receiver.getRemoteId()));
+
+        writer.append(buildDigestMsg(notifications, templateContext));
+      } catch (IOException e) {
+        ctx.setException(e);
+        return false;
+      }
+      return true;
     }
 
-  }
+    protected String buildDigestMsg(List<NotificationInfo> notifications, TemplateContext templateContext) {
+      Map<String, List<NotificationInfo>> map = new HashMap<>();
+      for (NotificationInfo notificationInfo : notifications) {
+        String notificationId = notificationInfo.getValueOwnerParameter(NotificationConstants.CONTEXT);
+        map.computeIfAbsent(notificationId, key -> new LinkedList<>()).add(notificationInfo);
+      }
+      StringBuilder sb = new StringBuilder();
+      for (Map.Entry<String, List<NotificationInfo>> notification : map.entrySet()) {
+        List<NotificationInfo> notificationInfos = map.get(notification.getKey());
```

Comment:
Instead of calling map, you should have something link 
notification.getValue()